core/storage/btree.rs

```diff
@@ -4846,12 +4787,6 @@ impl BTreeCursor {
                     let needs_balancing = leaf_underflows || interior_overflows_or_underflows;
 
                     if needs_balancing {
-                        let delete_info = self.state.mut_delete_info().unwrap();
```

Comment:
This was removed because we are sure that balance state will be `Start` here? Should we assert that here if yes?